docroot/modules/custom/mass_hierarchy/src/MassHierarchyRouteMatch.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Drupal\mass_hierarchy;
+
+use Drupal\Core\Routing\RouteMatchInterface;
+
+/**
+ * Custom RouteMatch class which includes setter method.
+ */
+class MassHierarchyRouteMatch implements RouteMatchInterface {
+
+  /**
```

Comment:
Would it make sense to only have setParameter() on this class? Other methods can inherit from parent. Some methods in this class look like copies of their parent (e.g. getParameters).

Reply:
Changed the implementation